src/Kestrel.Core/Internal/Http/Http1Connection.cs

```diff
@@ -274,13 +274,9 @@ private void OnAuthorityFormTarget(HttpMethod method, Span<byte> target)
 
             // This is not complete validation. It is just a quick scan for invalid characters
             // but doesn't check that the target fully matches the URI spec.
-            for (var i = 0; i < target.Length; i++)
+            if (HttpCharacters.IndexOfInvalidAuthorityChar(target) >= 0)
```

Comment:
You don't use the index anywhere.
`if (HttpCharacters.ContainsInvalidAuthorityChar(target))`

Reply:
Not in this case but it is used in the `HttpHeaders`: https://github.com/aspnet/KestrelHttpServer/blob/release/2.2/src/Kestrel.Core/Internal/Http/HttpHeaders.cs#L262. I wasn't sure if a mix of Contains and IndexOf is a good idea so I converted all of them into IndexOf which covers almost all the scenarios.

Reply:
Creating an API best suited to the caller is better than creating standard APIs for standards sake.